### PR TITLE
Add progress meter for Surrogate and LocalPermutation tests

### PR DESCRIPTION
**Pull request by Datseris**
Also simplifies the surrogate test source code. 

For parameteric tests I would prefer to do this at another point.

Partially addresses #341

### CODE REVIEW

**Comment by kahaaga** [comment]
Nice!! 

Just a few thoughts while I remember it:

- In the line `estimation = x -> estimate(measure, est, x, SSSets...)` you create a convenience function with only one input - the variable that should be shuffled, which is always assumed to be the first variable `x` in `independence(test::SurrogateTest, x, args...)`.  What happens if a user wants  to shuffle some of the other inputs? Or all inputs? We don't need to deal with this now, because it wasn't dealt with before this PR either, but it is something to keep in mind.

**Comment by Datseris** [comment]
Well in the current source code this wasn't a possibility anyways so the way I do it now is fine. In the future it could be generalized. It sholud be just `estimation = (x, y) -> estimate(measure, est, x, y, SSSets...)`.

Tests pass, can we merge and tag?

**Comment by kahaaga** [comment]
This looks good to me, but squashing and merging should wait until #350 is merged. The bug fix is more urgent than the progress indicator.

It will require some more work in this PR to accommodate the new transfer-entropy specific methods in #350.

**Review by kahaaga — APPROVED** [review]

**Pull request merged**
